File: Leetcode/sbgo/Leetcode724-寻找数组的中心索引.go
package sbgo

func pivotIndex(nums []int) int {
	n := len(nums)
	if n <= 2 {
		return -1
	}
	totalSum := 0
	for _, val := range nums {
		totalSum += val
	}
	sum := 0
	for i := range nums {
		// tmp := sum
		if sum == totalSum-sum-nums[i] {
			return i
		}
		sum += nums[i]
	}
	return -1
}

File: Leetcode/PY/Leetcode724-寻找数组的中心索引.py
from typing import List
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        if n <= 2:
            return -1
        totalSum = 0
        for num in nums:
            totalSum += num
        sum = 0
        for i in range(n):
            if totalSum - nums[i] - sum == sum:
                return i
            sum += nums[i]
        return -1